silo-core/contracts/incentives/base/BaseIncentivesController.sol

```diff
@@ -0,0 +1,299 @@
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function handleAction(
+        bytes32 _incentivesProgramId,
+        address _user,
+        uint256 _totalSupply,
+        uint256 _userBalance
+    ) public onlyNotifier {
+        uint256 accruedRewards = _updateUserAssetInternal(_incentivesProgramId, _user, _userBalance, _totalSupply);
+
+        if (accruedRewards != 0) {
+            uint256 newUnclaimedRewards = _usersUnclaimedRewards[_user][_incentivesProgramId] + accruedRewards;
+            _usersUnclaimedRewards[_user][_incentivesProgramId] = newUnclaimedRewards;
+
+            emit RewardsAccrued(
+                _user,
+                incentivesPrograms[_incentivesProgramId].rewardToken,
+                _incentivesProgramId,
+                newUnclaimedRewards
+            );
+        }
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function getRewardsBalance(address _user, string calldata _programName)
+        external
+        view
+        returns (uint256 unclaimedRewards)
+    {
+        bytes32 programId = getProgramId(_programName);
+        unclaimedRewards = getRewardsBalance(_user, programId);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function getRewardsBalance(address _user, bytes32 _programId)
+        public
+        view
+        returns (uint256 unclaimedRewards)
+    {
+        unclaimedRewards = _usersUnclaimedRewards[_user][_programId];
+
+        (uint256 stakedByUser, uint256 totalStaked) = _getScaledUserBalanceAndSupply(_user);
+
+        unclaimedRewards += _getUnclaimedRewards(_programId, _user, stakedByUser, totalStaked);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewards(address _to) external returns (AccruedRewards[] memory accruedRewards) {
+        if (_to == address(0)) revert InvalidToAddress();
+
+        accruedRewards = _accrueRewards(_to);
+        _claimRewards(msg.sender, msg.sender, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewards(address _to, bytes32[] calldata _programIds)
+        external
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        if (_to == address(0)) revert InvalidToAddress();
+
+        accruedRewards = _accrueRewardsForPrograms(_to, _programIds);
+        _claimRewards(msg.sender, msg.sender, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewards(address _to, string[] calldata _programNames)
+        external
+        returns (AccruedRewards[] memory accruedRewards)
+    {
+        if (_to == address(0)) revert InvalidToAddress();
+
+        bytes32[] memory programIds = _getProgramsIds(_programNames);
+        accruedRewards = _accrueRewardsForPrograms(_to, programIds);
+        _claimRewards(msg.sender, msg.sender, _to, accruedRewards);
+    }
+
+    /// @inheritdoc ISiloIncentivesController
+    function claimRewardsOnBehalf(address _user, address _to)
```

Comment:
We don't need a method that claims for all programs as long as we have a public getter to get and array of all programIds

Reply:
Done. Removed.